src/coreclr/jit/lsra.cpp

```diff
@@ -4668,15 +4707,14 @@ void LinearScan::freeRegisters(regMaskTP regsToFree)
     makeRegsAvailable(regsToFree);
     while (regsToFree != RBM_NONE)
     {
-        regMaskTP nextRegBit = genFindLowestBit(regsToFree);
-        regsToFree &= ~nextRegBit;
-        regNumber  nextReg   = genRegNumFromMask(nextRegBit);
+        regNumber nextReg = genFirstRegNumFromMaskAndToggle(regsToFree);
+
         RegRecord* regRecord = getRegisterRecord(nextReg);
 #ifdef TARGET_ARM
         if (regRecord->assignedInterval != nullptr && (regRecord->assignedInterval->registerType == TYP_DOUBLE))
         {
             assert(genIsValidDoubleReg(nextReg));
-            regsToFree &= ~(nextRegBit << 1);
+            regsToFree &= ~(genRegMask(nextReg) << 1);
```

Comment:
Is this better as this or as?
```
regsToFree &= ~genRegMask(nextReg + 1)`
```

It's a fairly minor difference between them and unsure off the top of my head which will be better codegen overall.

Likewise, are we wanting to explicitly clear the bit and its state is unknown so we have to do `and not`, or is its state known to be set and so we can do `^` instead?

Reply:
I think what you have suggested is correct. We know that it is a `double` interval and we need to clear the other half too.